include/depthimage_to_laserscan/DepthImageToLaserScan.h

```diff
@@ -64,8 +64,45 @@ namespace depthimage_to_laserscan
      * 
      */
     sensor_msgs::LaserScanPtr convert_msg(const sensor_msgs::ImageConstPtr& depth_msg,
-					   const sensor_msgs::CameraInfoConstPtr& info_msg);
+    	      const sensor_msgs::CameraInfoConstPtr& info_msg)
+    {
+    	sensor_msgs::LaserScanPtr scan_msg(new sensor_msgs::LaserScan());
+    	convert_msg(depth_msg, info_msg, scan_msg);
+    	return scan_msg;
+    }
+
+    /**
+     * Allocation-free converter
+     */
+    bool convert_msg(const sensor_msgs::ImageConstPtr& depth_msg,
+					   const sensor_msgs::CameraInfoConstPtr& info_msg, sensor_msgs::LaserScanPtr & scan_msg);
     
+
+    /**
+	 * Converts the information in a depth image (sensor_msgs::Image) to a sensor_msgs::LaserScan.
+	 *
+	 * This function converts the information in the depth encoded image (UInt16 or Float32 encoding) into
+	 * a sensor_msgs::LaserScan as accurately as possible.  To do this, it requires the synchornized Image/CameraInfo
+	 * pair associated with the image.
+	 *
+	 * This function uses float32 for most calculations. ARM/NEON platforms prefer float over double
+	 *
+	 * @param depth_msg UInt16 or Float32 encoded depth image.
+	 * @param info_msg CameraInfo associated with depth_msg
+	 * @return sensor_msgs::LaserScanPtr for the center row(s) of the depth image.
+	 *
+	 */
+    sensor_msgs::LaserScanPtr convert_msg_f(const sensor_msgs::ImageConstPtr& depth_msg,
```

Comment:
Inline here too. But I'd suggest that we don't need to add the wrapper on the allocation free conversion. Passing in the message is a cleaner API. And for the double one, it's important to keep it available for backwards compatibility, but since this is a new API we don't need to add the parallel interface.